packages/cli/src/workflow-runner.ts

```diff
@@ -424,6 +426,14 @@ export class WorkflowRunner {
 						data.workflowData,
 						{ retryOf: data.retryOf ? data.retryOf.toString() : undefined },
 					);
+
+					if (
+						error instanceof Error &&
+						error.message.includes('job stalled more than maxStalledCount')
+					) {
+						error = new MaxStalledCountError(error);
+					}
+
```

Comment:
can we move this block before line 421?

Reply:
What do we gain by doing this?

Reply:
I just find the order of things more intuitive and readable that way. error wrapping happens right after it is caught, and the hooks creation happens right before it is used.